src/pages/nfts/[address].tsx

```diff
@@ -222,6 +243,26 @@ const NftShow: NextPage<NftPageProps> = ({ marketplace }) => {
   const offer = find<Offer>(pipe(prop('buyer'), equals(publicKey?.toBase58())))(
     data?.nft.offers || []
   )
+  let activities: Activity[] = []
+  data?.nft.listings?.forEach((l: Listing) => {
+    activities.push({
+      type: ActivityType.Listed,
+      price: l.price,
+      fromWallet: l.seller,
+      createdAt: l.createdAt,
+    })
+  })
+  data?.nft.purchases?.forEach((p: Purchase) => {
+    activities.push({
+      type: ActivityType.Sold,
+      price: p.price,
+      fromWallet: p.seller,
+      toWallet: p.buyer,
+      createdAt: p.createdAt,
+    })
+  })
+  activities.sort((a, b) => +new Date(b.createdAt) - +new Date(a.createdAt))
+
```

Comment:
listings here is only active listing so we aren't getting any past listings.

What we want is custom query for pulling activities.

```
nft {
  activities {
    createdAt
    price
    type
    toWallet
    fromWallet
  }
}
```

Database wise this should start with querying the listing_receipts and joining in the purchase receipt. You can then make an NftActivity read model to map the record to a model.

We can go over tonight or tomorrow.

Reply:
@kespinola Ok, sure